cmd/kubectl-gadget/biolatency.go

```diff
@@ -0,0 +1,145 @@
+// Copyright 2019-2021 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/spf13/cobra"
+
+	"github.com/kinvolk/inspektor-gadget/cmd/kubectl-gadget/utils"
+	gadgetv1alpha1 "github.com/kinvolk/inspektor-gadget/pkg/api/v1alpha1"
+)
+
+const (
+	GADGET_NAME        = "biolatency"
+	GADGET_OUTPUTMODE  = "Status"
+	GADGET_OUTPUTSTATE = "Completed"
+)
+
+var biolatencyTraceConfig = &utils.TraceConfig{
+	GadgetName:       GADGET_NAME,
+	TraceOutputMode:  GADGET_OUTPUTMODE,
+	TraceOutputState: GADGET_OUTPUTSTATE,
+	CommonFlags:      &params,
+}
+
+var biolatencyCmd = &cobra.Command{
+	Use:   "biolatency",
+	Short: "Generate a histogram with the distribution of block device I/O latency",
+	PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
+		// Ensure parent's PersistentPreRunE is still called
+		if cmd.Parent() != nil && cmd.Parent().PersistentPreRunE != nil {
+			err := cmd.Parent().PersistentPreRunE(cmd.Parent(), args)
+			if err != nil {
+				return err
+			}
+		}
+
+		// Biolatency does not support filtering so we need to avoid adding
+		// the default namespace configured in the kubeconfig file.
+		if params.Namespace != "" && !params.NamespaceOverridden {
+			params.Namespace = ""
+		}
+
+		return nil
+	},
```

Comment:
According to my comment below this check is only needed for the start command and this checked can be done in `runBiolatencyStart`.

Reply:
Agree, I will apply this in case we decide to proceed in that way (Discussion [here](https://github.com/kinvolk/inspektor-gadget/pull/389#discussion_r747142238)).